node/chainSimulator/components/testOnlyProcessingNode.go

```diff
@@ -292,3 +295,51 @@ func (node *testOnlyProcessingNode) createNodesCoordinator(pref config.Preferenc
 
 	return nil
 }
+
+func (node *testOnlyProcessingNode) ProcessBlock(nonce uint64, round uint64) error {
+	bp := node.ProcessComponentsHolder.BlockProcessor()
+	newHeader, err := node.prepareHeader(nonce, round)
+	if err != nil {
+		return err
+	}
+
+	header, block, err := bp.CreateBlock(newHeader, func() bool {
+		return true
+	})
+	if err != nil {
+		return err
+	}
+
+	err = bp.ProcessBlock(header, block, func() time.Duration {
+		return 1000
```

Comment:
add a TODO to configure the round duration

Reply:
done